src/ProcessAbstract.php

```diff
@@ -105,7 +105,12 @@ public function exit(): void
     public function shutdown(): ProcessInterface
     {
         if ($this->_read(self::PROP_IS_SHUTDOWN_METHOD_ACTIVE)) {
-            $this->_getLogger()->critical("Shutdown method invoked.");
+            // to avoid hitting artificial memory limits while executing this block
+            ini_set('memory_limit','-1');
+            $this->_getLogger()->critical(
+                'Shutdown method invoked.',
+                ['error_get_last' => error_get_last()]
```

Comment:
```suggestion
                ['potentially_unrelated_error_get_last' => error_get_last()]
```

If the goal is reduce the potential of false correlation of `error_get_last()` with the shutdown, then I would encapsulate the level of skepticism one should use into the key name.

Reply:
fwiw the following reads to me like `**potentially_unrelated**` is a note on `error_get_last`
```php
['**potentially_unrelated**error_get_last' => error_get_last()]
```

Reply:
`error_get_last_albeit_potentially_not_an_uncaught_error_that_caused_the_shutdown`